packages/legend-query/src/stores/QueryBuilderState.ts

```diff
@@ -382,6 +391,10 @@ export class QueryBuilderState {
       const compiledLambda = guaranteeNonNullable(
         compiledValueSpecification.values[0],
       );
+      observe_ValueSpecification(
```

Comment:
Why don't we just observe `valueSpec` as we created it instead?

Reply:
Can you elaborate. We create it during the buiilder. I added it directly after building the value spec.